### PR TITLE
feature/readwise integration api

### PR DESCRIPTION
**Pull request by sywhb**
- Add integrations table in db
- Add integration entity
- Add createIntegration API to the GQL schema
- Add createIntegration API implementation

### CODE REVIEW

**Comment by sywhb** [comment]
#999

**Comment by sywhb** [comment]
Adding integration tests...

File: packages/db/migrations/0091.do.integrations.sql [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+-- Description: Create integrations table
+
+BEGIN;
+
+CREATE TYPE omnivore.integration_type AS ENUM ('READWISE');
+
+CREATE TABLE omnivore.integrations (
+    id uuid PRIMARY KEY DEFAULT uuid_generate_v1mc(),
+    user_id uuid NOT NULL REFERENCES omnivore.user ON DELETE CASCADE,
+    "type" omnivore.integration_type NOT NULL,
```
Comment (jacksonh) [inline packages/db/migrations/0091.do.integrations.sql:12]: 👍

File: packages/api/src/resolvers/integrations/index.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+  try {
+    const user = await getRepository(User).findOneBy({ id: uid })
+    if (!user) {
+      return {
+        errorCodes: [SetIntegrationErrorCode.Unauthorized],
+      }
+    }
+
+    const integrationToSave: Partial<Integration> = {
```
Comment (jacksonh) [inline packages/api/src/resolvers/integrations/index.ts:32]: Do we need to sanitize these input values? I guess GQL kind of does already, and then the DB will verify that its a valid type.

File: packages/api/src/services/integrations.ts [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+): Promise<boolean> => {
+  switch (type) {
+    case IntegrationType.Readwise:
+      return validateReadwiseToken(token)
+    default:
+      return false
+  }
+}
+
+const validateReadwiseToken = async (token: string): Promise<boolean> => {
```
Comment (jacksonh) [inline packages/api/src/services/integrations.ts:51]: Good idea validating right here.

**Review by jacksonh — APPROVED** [review]

**Comment by jacksonh** [comment]
I guess we will have to do some testing on demo.

**Review by jacksonh — APPROVED** [review]

**Pull request merged**